mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -496,22 +517,21 @@ private void assertIsOwner(MailboxSession mailboxSession, MailboxPath mailboxPat
         }
     }
 
-    private void doRenameMailbox(MailboxPath from, MailboxPath to, MailboxSession session, MailboxMapper mapper) throws MailboxException {
+    private void doRenameMailbox(Mailbox mailbox, MailboxPath newMailboxPath, MailboxSession session, MailboxMapper mapper) throws MailboxException {
         // TODO put this into a serilizable transaction
-        Mailbox mailbox = Optional.ofNullable(mapper.findMailboxByPath(from))
-            .orElseThrow(() -> new MailboxNotFoundException(from));
 
-        mailbox.setNamespace(to.getNamespace());
-        mailbox.setUser(to.getUser());
-        mailbox.setName(to.getName());
+        mailbox.setNamespace(newMailboxPath.getNamespace());
+        mailbox.setUser(newMailboxPath.getUser());
+        mailbox.setName(newMailboxPath.getName());
         mapper.save(mailbox);
 
+        MailboxPath from = mailbox.generateAssociatedPath();
```

Comment:
you still have a race condition then, no?

Reply:
Maybe in the result returned to the client (new name) but not on the store content...